### PR TITLE
DB-10597 Support DB2 style varchar comparison.

### PR DESCRIPTION
**Pull request by msirek**
Comparison of trailing spaces in VARCHAR is ignored with the following database property enabled:
    call syscs_util.syscs_set_global_database_property('**splice.db2.varchar.compatible**', true);

### CODE REVIEW

**Comment by msirek** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1836/

**Comment by msirek** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/iapi/types/DataValueFactoryImpl.java [hunk 1]
```diff
@@ -1290,6 +1290,7 @@ public static DataValueDescriptor getNullDVDWithUCS_BASICcollation(
             case StoredFormatIds.SQL_TIMESTAMP_ID: return new SQLTimestamp();
             case StoredFormatIds.SQL_TINYINT_ID: return new SQLTinyint();
             case StoredFormatIds.SQL_VARCHAR_ID: return new SQLVarchar();
+            case StoredFormatIds.SQL_VARCHAR_DB2_COMPATIBLE_ID: return new SQLVarcharDB2Compatible();
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/iapi/types/DataValueFactoryImpl.java:1293]: Do we need to add `case StoredFormatIds.SQL_VARCHAR_DB2_COMPATIBLE_ID` for getNullDVD() also?

**Review by carolp-503 — APPROVED** [review]
+1

File: db-engine/src/main/java/com/splicemachine/db/iapi/types/DataValueFactoryImpl.java (hunk 1 above)
Comment (msirek) [reply db-engine/src/main/java/com/splicemachine/db/iapi/types/DataValueFactoryImpl.java:1293]: I don't think so.  The only place EngineUtils.getFormatIds is called is in SparkValueRowSerializer.write, and the only place this is consumed is in SparkValueRowSerializer.read:

>                 formatIds[i] = input.readInt(true);
>                 rowTemplate[i] = getDVD(formatIds[i]);

Also, getNullDVD deals with different typeFormatIds (the ones that don't start with "SQL_").

Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/iapi/types/DataValueFactoryImpl.java:1293]: I see, thanks Mark!

**Review by yxia92 — APPROVED** [review]
+1

**Review by jyuanca — APPROVED** [review]

**Pull request merged**